### PR TITLE
Feature/kas 3884 beslissingen downloaden

### PR DESCRIPTION
**Pull request by StuyckRuben**
https://kanselarij.atlassian.net/browse/KAS-3884

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -11,10 +11,17 @@ app.post('/agendas/:agenda_id/agendaitems/documents/files/archive', async (req,
   let files;
   if (mandateeIdsString) {
     const mandateeIds = mandateeIdsString.split(',');
+    if (req.query.decisions){
+      files = await fetchDecisionsByMandatees(req.params.agenda_id, mandateeIds)
+    }
     files = await fetchFilesFromAgendaByMandatees(req.params.agenda_id, mandateeIds);
```
Comment (sergiofenoll) [inline app.js]: Needs an else because you're overwriting the fetched files here.

File: app.js [hunk 2]
```diff
@@ -11,10 +11,17 @@ app.post('/agendas/:agenda_id/agendaitems/documents/files/archive', async (req,
   let files;
   if (mandateeIdsString) {
     const mandateeIds = mandateeIdsString.split(',');
+    if (req.query.decisions){
```
Comment (sergiofenoll) [inline app.js]: I suspect `req.query.decisions` will contain the string `true` or `false` (given that `URLSearchParams` is correctly used in the frontend). You will want to explicitly check `if (req.query.decisions === 'true')` here.

**Review by sergiofenoll — CHANGES_REQUESTED** [review]
Needs some small changes, see comments.

File: app.js (hunk 2 above)
Comment (ValenberghsSven) [reply app.js]: fixed

File: app.js (hunk 1 above)
Comment (ValenberghsSven) [reply app.js]: fixed

**Comment by ValenberghsSven** [comment]
> Needs some small changes, see comments.

I had to change the queries with mandatees.
Running them manually returned an error `Variable 'agendaitem' is used in subexpressions of the query but not assigned`
Resource seemed fine handling it regardless of that?

**Comment by ValenberghsSven** [comment]
I made an attempt to create 1 query with blocks.
It brought me too far into the weeds and in the end I just changed the existing queries.
Perhaps it would be better to separate some parts and select the agendaitems (with or without mandatees)
Then pass that list to a different queries that just gets all the documents or decisions from the list of selected agendaitems.
In a follow up story I mean.

**Review by sergiofenoll — APPROVED** [review]

**Pull request merged**